### PR TITLE
Cmake & minor fixes

### PR DESCRIPTION
**Pull request by klemens-morgenstern**

### CODE REVIEW

File: mrdox.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+project-name: MrDox
+namespaces:
+  include:
+    - ::N
```
Comment (vinniefalco) [inline mrdox.yml:4]: Now this is interesting! There's missing newlines by the way. And if you are going to add this mrdox.yml please put it in its own folder.

File: CMakeLists.txt [hunk 2]
```diff
@@ -14,7 +14,7 @@
 #
 #-------------------------------------------------
 
-cmake_minimum_required(VERSION 3.25.2)
+cmake_minimum_required(VERSION 3.21)
```
Comment (vinniefalco) [inline CMakeLists.txt:17]: This seems ok

File: source/lib/Config.cpp [hunk 3]
```diff
@@ -102,7 +103,7 @@ loadFromFile(
     auto fileText = llvm::MemoryBuffer::getFile(configPath);
     if(! fileText)
     {
-        R.failed("llvm::MemoryBuffer::getFile", fileText);
+        R.failed("llvm::MemoryBuffer::getFile", configPath);
```
Comment (vinniefalco) [inline source/lib/Config.cpp:106]: There's two "include" file filters now?

File: include/mrdox/Corpus.hpp [hunk 4]
```diff
@@ -118,8 +118,8 @@ class Corpus
 
     /** Build the intermediate representation of the code being documented.
     */
-    static
     [[nodiscard]]
+    static
```
Comment (vinniefalco) [inline include/mrdox/Corpus.hpp:122]: A fine change

File: source/lib/Serialize.cpp [hunk 5]
```diff
@@ -478,7 +478,7 @@ populateParentNamespaces(
             InfoType::IT_namespace);
 }
 
-void PopulateTemplateParameters(std::optional<TemplateInfo>& TemplateInfo,
+void PopulateTemplateParameters(llvm::Optional<TemplateInfo>& TemplateInfo,
```
Comment (vinniefalco) [inline source/lib/Serialize.cpp:481]: Why don't we just replace it everywhere?

File: include/mrdox/Generator.hpp [hunk 6]
```diff
@@ -66,7 +66,8 @@ struct Generator
         llvm::StringRef rootPath,
         Corpus const& corpus,
         Config const& config,
-        Reporter& R) const;
+        Reporter& R,
+        const std::string & fileName) const;
```
Comment (vinniefalco) [inline include/mrdox/Generator.hpp:70]: The `Reporter` always comes last in parameter lists but why are you changing the API of `Generator`?

File: source/tool/ToolMain.cpp [hunk 7]
```diff
@@ -147,7 +155,7 @@ toolMain(
 
     // Run the generator.
     llvm::outs() << "Generating docs...\n";
-    if(! gen->build(config.OutDirectory, *rv, config, R))
+    if(! gen->build(config.OutDirectory, *rv, config, R, Filename))
```
Comment (vinniefalco) [inline source/tool/ToolMain.cpp:158]: This is not right though, because the non-XML generators emit multi-file output

Comment (klemens-morgenstern) [reply source/tool/ToolMain.cpp:158]: does the non-xml have a primary file I can use? I need some target for cmake.

**Comment by vinniefalco** [comment]
I merged some of these commits, please rebase

File: source/lib/Serialize.cpp (hunk 5 above)
Comment (klemens-morgenstern) [reply source/lib/Serialize.cpp:481]: I did.

**Comment by vinniefalco** [comment]
Is this obsolete?

**Pull request closed without merging**